networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -147,30 +150,61 @@ where
 			}
 		}
 
-		let length = transactions.len();
-		if length > 0 {
-			let mut light_node_client = self.light_node_client.clone();
-			let span = info_span!(
+		if transactions.len() > 0 {
+			info!(
 				target: "movement_timing",
-				"batch_write",
 				batch_id = %batch_id,
-				length = length
+				transaction_count = transactions.len(),
+				"built_batch_write"
 			);
-			light_node_client
-				.batch_write(BatchWriteRequest { blobs: transactions })
-				.instrument(span)
-				.await?;
-			// We now consider the transactions no longer in mempool flight.
-			self.executor.decrement_transactions_in_flight(length as u64);
+			let batch_write = BatchWriteRequest { blobs: transactions };
+			sender.send(batch_write).await?;
 		}
 
 		Ok(())
 	}
 
+	pub async fn send_transaction_batch_writes(
+		&self,
+		sender: Sender<BatchWriteRequest>,
+	) -> Result<(), anyhow::Error> {
+		loop {
+			self.next_transaction_batch_write(sender.clone()).await?;
+		}
+	}
+
+	pub async fn submit_transaction_batch_writes(
+		&self,
+		receiver: Receiver<BatchWriteRequest>,
+	) -> Result<(), anyhow::Error> {
+		loop {
+			let batch_write = receiver.recv().await?;
+			let mut light_node_client = self.light_node_client.clone();
+			// batch_writes can be submitted in parallel without waiting for the previous one to finish
+			tokio::spawn(async move {
+				light_node_client.batch_write(batch_write).await?;
+				Ok::<(), anyhow::Error>(())
+			});
+		}
+	}
+
 	pub async fn write_transactions_to_da(&self) -> Result<(), anyhow::Error> {
 		loop {
-			self.tick_write_transactions_to_da().await?;
+			let (batch_sender, batch_receiver) = async_channel::unbounded();
+
+			// run send transaction batch writes and submit transaction batch writes concurrently
+			match tokio::try_join!(
+				self.send_transaction_batch_writes(batch_sender),
+				self.submit_transaction_batch_writes(batch_receiver)
+			) {
```

Comment:
So these two futures communicate over the (unnecessarily over-synchronized) channel, but are driven preemptively as one task by `try_join!`. I think this is just useless overhead now.

If you spawn these two as tasks, you may get real benefits on the multithreaded executor. But this would require state for each of them to be taken out of `self`, so it might be a taller order.

Reply:
The benefit here is that the batches keep building without waiting for the submission of the `BatchWriteRequest` to occur. This is not intended for optimization viz the runtime, but more logical optimization w.r.t. the the throughput of the `m1-da-light-node` service.

Reply:
Also, I don't think there's anything particularly CPU bound here. So, what would be the benefit? It's basically waiting on a channel--which the futures as coroutines should be good at--and sending requests to `m1-da-light-node` which coroutines should also be good at.

To my knowledge, spawn can be better for more CPU-intensive tasks because it may spawn an additional green thread for the task. However, shouldn't the benefit be negligible here?

Reply:
Yes, the futures polled concurrently are still beneficial, but this can be done without a channel (which is now just a buffer with synchronization overhead), e.g. with `select!`.

Reply:
I'm not seeing how `select!` replaces anything the current design. I think I would effectively need to `select!`, `join_all!`, or `join_each_concurrent` on a dynamic join set. I usually opt for a channel and a worker over trying something exotic for that kind of pattern

However, if you don't want the channel, I can just spawn and not await the submission task from the batching loop.

Reply:
I thought of having a loop with `select!` in the body where one `Stream` object would yield batches and another future consume the available batch.

But if it's easier to do over a channel now, let it be.